index.js

```diff
@@ -92,16 +98,38 @@ Client.prototype.write = function(msg, tags){
     msg += "|#" + tags.join(',');
   }
 
+  // Exceeds maximum size.
   if (this.bufferSize && this.bufferSize > 0) {
     if ((this.buffer.length + msg.length) > this.bufferSize) {
       this.flush();
     }
     this.buffer += msg;
     this.buffer += '\n';
-    return;
+    alreadyAddedToBuffer = true;
+    doNotSend = true;
   }
 
-  this.send(msg);
+  // Exceeds flushAfter timeout
+  if (this.flushAfter && this.flushAfter > 0) {
+    // We know we're under the max buffer size.
+    if (!alreadyAddedToBuffer) {
+      this.buffer += msg;
+      alreadyAddedToBuffer = true;
+    }
+
+    if (!this.flushScheduled) {
+      var self = this; // Assuming arrow functions are a no-no.
+      setTimeout(function(){
```

Comment:
@achille-roussel Not familiar with setTimer specifically, do you have an example? :)

Reply:
Ah! nevermind my JS is rusty, I meant `setInterval`

Reply:
Ahh, definitely! I wasn't sure if I wanted to just set a global setInterval call or not, since that would continue running even if there were no calls to flush, which is only a potential performance concern. I'm happy to do either though, it's more a question of style!